### PR TITLE
Fix undefined declrations

### PR DESCRIPTION
**Pull request by dvoytenko**
/cc @cramforce

### CODE REVIEW

**Comment by erwinmombay** [comment]
@dvoytenko LGTM

**Comment by cramforce** [comment]
Resource class has a couple I believe.

Is there a lint check for this by any chance?

**Comment by erwinmombay** [comment]
@cramforce there is the "unused expressions" check which can catch these, but chai also triggers those right now because of the `expect(blah)to.be.blah` style. (which is one of the reasons i want to get rid of it)

File: src/custom-element.js [hunk 1]
```diff
@@ -305,13 +305,13 @@ export function createAmpElementProto(win, name, implementationClass) {
...
 
     /** @private {string|null|undefined} */
-    this.mediaQuery_;
+    this.mediaQuery_ = undefined;
 
     /** @private {!SizeList|null|undefined} */
-    this.sizeList_;
+    this.sizeList_ = undefined;
 
     /** @private {!SizeList|null|undefined} */
```
Comment (erwinmombay) [inline src/custom-element.js]: couldnt we just assign `null` as the default then to simplify the types on these? (im ok w/ not changing these if it affects the actual code)

Comment (erwinmombay) [inline src/custom-element.js]: nvm, looks like explicitly check for `undefined` in comparisons.

**Comment by dvoytenko** [comment]
@cramforce looks like i only missed one.

File: src/custom-element.js (hunk 1 above)
Comment (dvoytenko) [inline src/custom-element.js]: correct, `undefined` is a meaningful value here.

**Comment by dvoytenko** [comment]
@erwinmombay ptal. let's push this through and then feel free to enable linter.

**Comment by erwinmombay** [comment]
@dvoytenko LGTM

**Pull request merged**